### DISCUSSION TITLE
Undo Close Tab

### DISCUSSION THREAD

**Discussion by cryham** [doublecmd/doublecmd#1093] [Suggestions]
### Description

I already added it back in 2019 (partly)
https://doublecmd.sourceforge.io/mantisbt/view.php?id=2032

Was already suggested in 2017,
https://doublecmd.h1n.ru/viewtopic.php?f=7&t=4018&p=21213&hilit=undo+close+tab#p21213
in 2018 by me,
in 2014:
https://doublecmd.sourceforge.io/mantisbt/view.php?id=942
and in 2019:
https://doublecmd.h1n.ru/viewtopic.php?p=26416#p26416

For me this is still the number 1 missing feature in DC.
I'm curious if you could just add it? And if anything changed in your view on this.

It's been about 3,5 years since I made my fork (in 2019) and simply didn't have the time (or desire) to update, I surely don't have it now either.
More about my fork here: https://cryham.tuxfamily.org/portfolio/2019-doublecmd-fork/#Sources
Link has also other features, some to look cooler (for me), but few I find also badly missing, like:
coloring file types in Find Files and in Multi Rename, which were easy to do.

**Comment by LIMUNADE** [comment]
I was totally convinced that dc has undo command option until one of my friends told me that he can't undo things. Then I searched an option in hotkeys and noticed that ctrl + z using "edit comment" option instead. I mean, this is very important feature. How come that since 2012 it didn't change at all?

**Reply by j2969719** [reply]
this is an important feature for navigational file managers like explorer, while double commander is an orthodox file manager. like most of the norton commander successors, it does not provide this feature. whats so surprising about this?

**Comment by cryham** [comment]
What's surprising is that since years it hasn't been implemented, while there have been few request already. Apparently it's not important enough for core developers or contributors.
Doesn't matter what DC is, any Gui tab control, where you can add tabs, and close them (by mistake too) is nice to have an undo close tab too. And shouldn't be a problem to implement (not a point for discussion IMO).
E.g. Firefox (and others) have this feature and DC could too.

**Reply by ags1234** [reply]
I completely agree to this.

**Comment by j2969719** [comment]
there were also few requests here to make the dc single-panel or convert ui to ribons.. would it be nice to have every feature imaginable? sure, but does this project realistically have enough man power to also implement every non-essential one?  and is it correct to compare a niche file manager team of two active developers and several contributors with a mainstream web browser team?

**Comment by cryham** [comment]
IDC whatever. Like I wrote, I implemented it myself years ago, and I use it few times daily since. It was agreed as optional feature. It wasn't perfect and I didn't have time, so it didn't go into main DC and is in my old fork. Basing on how this one feature was received I abandoned contributing any other features I made in my DC fork. I got other projects.

**Comment by alexx2000** [comment]
Technically page is not closing in your implementation, it is hiding instead. So it is still consume resources (not large of course), that I don't like personally. But I can add this feature as option.

**Comment by cryham** [comment]
Yes it is buggy. It also has a problem, I'm not sure when, but when I close a tab and undo close then I get an invisible tab, which can be focused if I cycle into it with Shift-Tab or Tab. It behaves badly, but I got used to it. Closing DC and opening fixes it for current tabs. I thought maybe someone could code it properly (instead of my buggy implementation), since I don't have knowledge of Lazarus and Pascal and DC code is rather big.
